### PR TITLE
Version header for proof hints

### PR DESCRIPTION
**Pull request by gtrepta**
fixes: #883

### CODE REVIEW

File: docs/proof-trace.md [hunk 1]
```diff
@@ -21,7 +21,9 @@ have rendered this unnecessary, but the change hasn't been implemented yet.
 
 Here is a BNF styled description of the format:
 ```
-proof_trace ::= event*
+proof_trace ::= header event*
+
+header      ::= "HINT" <4-byte version number>
```
Comment (Baltoli) [inline docs/proof-trace.md:26]: Could we have this be a semantic version like in the binary KORE format (which uses 3 16-bit unsigned integers for `major . minor . patch`)

File: runtime/util/util.cpp [hunk 2]
```diff
@@ -28,4 +28,13 @@ block *constructKItemInj(void *subject, const char *sort, bool raw_value) {
   auto args = std::vector<void *>{add_indirection ? (void *)&subject : subject};
   return static_cast<block *>(constructCompositePattern(tag, args));
 }
+
+void printProofHintHeader(char *output_file) {
+  unsigned int version = 1;
```
Comment (Baltoli) [inline runtime/util/util.cpp:33]: https://github.com/runtimeverification/llvm-backend/blob/master/include/kllvm/binary/version.h maybe pull this class upwards out of the `binary` library and reuse it here?

File: docs/proof-trace.md (hunk 1 above)
Comment (Baltoli) [reply docs/proof-trace.md:26]: See comment below, in fact

Comment (gtrepta) [reply docs/proof-trace.md:26]: @dwightguth asked for the single integer over the semantic version. I think I agree with doing it this way, the format shouldn't really go through any radical changes, and there probably wont be any widespread use of it outside of a couple of our tools.

Comment (Baltoli) [reply docs/proof-trace.md:26]: In that case it's fine by me!

**Review by Baltoli — APPROVED** [review]

**Pull request merged**